vmdb/app/models/orchestration_stack.rb

```diff
@@ -40,25 +40,39 @@ def total_cloud_networks
     cloud_networks.size
   end
 
-  # @param options [Hash] what to update for the stack. Option keys and values are:
-  #   :template (String, URI, S3::S3Object, Object) - A new stack template.
-  #     This may be provided in a number of formats including:
-  #       a String, containing the template in CFN or HOT format.
-  #       a URL String pointing to the document in S3.
-  #       a URI object pointing to the document in S3.
-  #       an S3::S3Object which contains the template.
-  #       an Object which responds to #to_json and returns the template.
-  #   :parameters (Hash) - A hash that specifies the input parameters of the new stack.
+  def self.create_stack(orchestration_manager, stack_name, template, options = {})
+    klass = "OrchestrationStack#{orchestration_manager.class.name.match(/Ems(.*)/)[1]}".constantize
```

Comment:
The reason I prefer a naming convention rather than a `kind_of?` switch is we don't need to come back to add new subclass. Currently we only have `OrchestrationStackAmazon` and `OrchestrationStackOpenstack`. In the future there is a higher possibility of adding an `OrchestrationStackAbc` than refactoring again EMS providers or managers.